kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/generic/QuectelGeneric.java

```diff
@@ -165,6 +173,131 @@ public ModemRegistrationStatus getRegistrationStatus() throws KuraException {
+        }
+        return this.asQnwinfo;
+    }
+
+    public void getExtendedRegistrationStatus() throws KuraException {
+        synchronized (this.atLock) {
+            String atPort = getAtPort();
+
+            logger.debug("sendCommand getLac :: {}", QuectelGenericAtCommands.GET_REGISTRATION_STATUS.getCommand());
+            byte[] reply;
+            CommConnection commAtConnection = openSerialPort(atPort);
+            if (!isAtReachable(commAtConnection)) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+            }
+            try {
+                commAtConnection.sendCommand(QuectelGenericAtCommands.GET_EXTENDED_REGISTRATION_STATUS.getCommand().getBytes(),
+                        1000, 100);
+                reply = commAtConnection.sendCommand(QuectelGenericAtCommands.GET_REGISTRATION_STATUS.getCommand().getBytes(),
+                        1000, 100);
+            } catch (IOException e) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.CONNECTION_FAILED, e);
+            }
+            closeSerialPort(commAtConnection);
+            if (reply != null) {
+                String sCgreg = this.getResponseString(reply);
+                getExtendedRegistrationStatusReply(sCgreg);
+            }
+        }
+    }
+
+    public String[] getExtendedRegistrationStatusReply(String sCgreg) throws KuraException {
+        int sLac;
+        int sCi;
+        if (sCgreg.startsWith("+CGREG:")) {
+            sCgreg = sCgreg.substring("+CGREG:".length()).trim();
+            logger.trace("getLac() :: +CGREG={}", sCgreg);
+            asCgreg = sCgreg.split(",");
+            sLac = Integer.parseInt(asCgreg[2].substring(1, asCgreg[2].length() - 1), 16);
+            sCi = Integer.parseInt(asCgreg[3].substring(1, asCgreg[3].length() - 1), 16);
+            this.lac = String.valueOf(sLac);
+            this.ci = String.valueOf(sCi);
+        }
+        return asCgreg;
+    }
+
+    @Override
+    public void getRegisteredNetwork() throws KuraException {
+        synchronized (this.atLock) {
+            String atPort = getAtPort();
+
+            logger.debug("sendCommand getRegisteredNetwork :: {}", QuectelGenericAtCommands.GET_REGISTERED_NETWORK.getCommand());
+            byte[] reply;
+            CommConnection commAtConnection = openSerialPort(atPort);
+            if (!isAtReachable(commAtConnection)) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+            }
+            try {
+                reply = commAtConnection.sendCommand(QuectelGenericAtCommands.GET_REGISTERED_NETWORK.getCommand().getBytes(),
+                        1000, 100);
+            } catch (IOException e) {
+                closeSerialPort(commAtConnection);
```

Comment:
we could close connection in a finally statement

Reply:
OK